lib/core.js

```diff
@@ -431,15 +439,6 @@ declare class Promise<R> {
     static reject<T>(error?: any): Promise<T>;
     static all<T, Elem: Promise<T> | T>(promises: Array<Elem>): Promise<Array<T>>;
     static race<T, Elem: Promise<T> | T>(promises: Array<Elem>): Promise<T>;
-
-    // Non-standard APIs common in some libraries
```

Comment:
> it's that it might not exist at all

This is a real problem and something we're in a good position to address so I agree with your basis, but it's no different an issue than having APIs that match a different version of Node than what you're using or an ES6 API that you aren't polyfilling/isn't supported by some platforms yet.

We do need better ways of toggling these kinds of things but "copy/paste the promise libdef and hack it with your own changes" is a pretty unfortunate one. Especially considering how pervasive `done()` is and the fickleness of typing `Promise` APIs. We seem to tweak and refine Promise a lot in ways that often aren't very intuitive (i.e. the need for things like `Promise<T> | T`...which is still not even quite right).